myapp/utils/chains.py

```diff
@@ -0,0 +1,206 @@
+
+
+class feature(BaseModel):
+    features: str = Field(
+        description="The specific features or use cases discussed by the client in the estimation sheet"
+    )
+    sub_functionality_of_features: List["str"] = Field(
+        description="It will include the sub functionality it will require for its complete implemenation. try to add all possible subfunctionality any kind of third party integartions etc"
+    )
+    UI_pages: List["str"] = Field(
+        description="It will include the all the frontend pages we need to make these models. Try to cover every sub functionality in these front-end pages"
+    )
+    assumption: str = Field(
+        descripion="make assumption about how it will be implemented and what technologies or third party requirement we will be needing"
+    )
+    backend_engineer: int = Field(
+        ...,
+        description="Number of backend engineer is assigned to work on this feature. Example: '0','1','2'etc.",
+    )
+
+    frontend_engineer: int = Field(
+        ...,
+        description="NUmber of  frontend engineer is assigned to work on this feature. Example: '0','1', '2' etc.",
+    )
+
+    QA: int = Field(
+        ...,
+        description="Number of QA engineer is assigned to test this feature. Example: '0','1','2'.",
+    )
+    AI_Engineer: int = Field(
+        ...,
+        description="Number of AI engineer needed if there is any task related to AI. Example: '0','1','2'.",
+    )
+
+
+class features_list(BaseModel):
+    features_list: List[feature]
+
+
+def create_qa_pair_chain():
+    model = ChatOpenAI(temperature=0, model="gpt-4o-mini")
+
+    parser = PydanticOutputParser(pydantic_object=QAPair)
+    prompt = PromptTemplate(
+        template=qa_pair_template,
+        input_variables=["requirement_document", "user_queries"],
+        partial_variables={"format_instructions": parser.get_format_instructions()},
+    )
+    return prompt | model | parser
+
+
+qa_pair_generator_chain = create_qa_pair_chain()
+
+
+def create_phase_extractor_chain():
+    model = ChatOpenAI(temperature=0, model="gpt-4o-mini")
+
+    parser = PydanticOutputParser(pydantic_object=phase_list)
+    prompt = PromptTemplate(
+        template=phase_template,
+        input_variables=["requirement_document", "user_queries"],
+        partial_variables={"format_instructions": parser.get_format_instructions()},
+    )
+    return prompt | model | parser
+
+
+phase_extractor_chain = create_phase_extractor_chain()
+
+
+def create_feature_extractor_chain():
+    model = ChatOpenAI(temperature=0, model="gpt-4o-mini")
+
+    parser = PydanticOutputParser(pydantic_object=features_list)
+    prompt = PromptTemplate(
+        template=feature_template,
+        input_variables=["est_sheet", "feedback", "user_queries"],
+        partial_variables={"format_instructions": parser.get_format_instructions()},
+    )
+    return prompt | model | parser
+
+
+feature_extractor_chain = create_feature_extractor_chain()
```

Comment:
This should not be here. call this function inside class, functions like inside your API.